README.md

```diff
@@ -1,24 +1,172 @@
-# README
+<a name="readme-top"></a>
 
-This README would normally document whatever steps are necessary to get the
-application up and running.
+<div align="center">
 
-Things you may want to cover:
+  <h3><b>Hello Rails and React</b></h3>
 
-* Ruby version
+</div>
 
-* System dependencies
+<!-- TABLE OF CONTENTS -->
 
-* Configuration
+# 📗 Table of Contents
 
-* Database creation
+- [📖 About the Project](#about-project)
+  - [🛠 Built With](#built-with)
+    - [Tech Stack](#tech-stack)
+    - [Key Features](#key-features)
+- [💻 Getting Started](#getting-started)
+  - [Prerequisites](#prerequisites)
+  - [Setup](#setup)
+  - [Install](#install)
+  - [Usage](#usage)
+  - [Run tests](#run-tests)
+  - [Deployment](#deployment)
+- [👥 Authors](#authors)
+- [🔭 Future Features](#future-features)
+- [🤝 Contributing](#contributing)
+- [⭐️ Show your support](#support)
+- [🙏 Acknowledgements](#acknowledgements)
+- [📝 License](#license)
 
-* Database initialization
+<!-- PROJECT DESCRIPTION -->
 
-* How to run the test suite
+# 📖 Hello Rails&React <a name="about-project"></a>
 
-* Services (job queues, cache servers, search engines, etc.)
+**Hello Rails&React** is project built with Rails, React and Redux, it is actually for learning purpose,
+it has a feature of displaying greetings randomly
 
-* Deployment instructions
+## 🛠 Built With <a name="built-with"></a>
 
-* ...
+### Tech Stack <a name="tech-stack"></a>
+
+<details>
+  <summary>Client</summary>
+  <ul>
+    <li><a href="#">Rails</a></li>
+    <li><a href="#">React</a></li>
+    <li><a href="#">Redux</a></li>
+    <li><a href="#">webpack</a></li>
+  </ul>
+</details>
+
+<details>
+<summary>Database</summary>
+  <ul>
+    <li><a href="https://www.postgresql.org/">PostgreSQL</a></li>
+  </ul>
+</details>
+
+<!-- Features -->
+
+### Key Features <a name="key-features"></a>
+
+>
+
+- **Generating Random Greetings**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- GETTING STARTED -->
+
+## 💻 Getting Started <a name="getting-started"></a>
+
+To get a local copy up and running, follow these steps.
+
+### Prerequisites
+
+In order to run this project you need to :
+
+- To have Ruby installed
+- To have Rails 7 installed also
+- Dependencies for basic gems that project uses
+
+### Setup
+
+Clone this repository to your desired folder:
+
+```sh
+  cd my-folder
+  git clone https://github.com/Peter-Dumbari/Hello-Rails-React.git
+
+```
+
+### Install
+
+Install this project with:
+
+```sh
+  cd my-project
+```
+
+### Usage
+
+To run the project, execute the following command:
+
+```sh
+  rails server
+```
+
+### Run tests
+
+To run tests, run the following command:
+
+- For eslint run the following:
+
+```sh
+  rubocop
+```
+
+### Deployment
+
+- Heroku etc.
+
```

Comment:
- [ ] Kindly make sure to remove the `Deployment` section as long as your project is not deployed. KIndly remove this section in order to have a clean and professional Readme.